llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp

```diff
@@ -496,15 +549,32 @@ void Dwarf5AccelTableWriter::emitAbbrevs() const {
 }
 
 void Dwarf5AccelTableWriter::emitEntry(
-    const DWARF5AccelTableData &Entry) const {
+    const DWARF5AccelTableData &Entry,
+    const DenseMap<uint64_t, MCSymbol *> &DIEOffsetToAccelEntryLabel,
+    DenseSet<MCSymbol *> &EmittedAccelEntrySymbols) const {
   std::optional<DWARF5AccelTable::UnitIndexAndEncoding> EntryRet =
       getIndexForEntry(Entry);
-  uint32_t AbbrvTag = constructAbbreviationTag(Entry.getDieTag(), EntryRet);
+  std::optional<uint64_t> MaybeParentOffset = Entry.getParentDieOffset();
+  std::optional<dwarf::Form> MaybeParentForm =
+      getFormForIdxParent(IndexedOffsets, MaybeParentOffset);
+  uint32_t AbbrvTag =
+      constructAbbreviationTag(Entry.getDieTag(), EntryRet, MaybeParentForm);
   auto AbbrevIt = Abbreviations.find(AbbrvTag);
   assert(AbbrevIt != Abbreviations.end() &&
          "Why wasn't this abbrev generated?");
   assert(getTagFromAbbreviationTag(AbbrevIt->first) == Entry.getDieTag() &&
          "Invalid Tag");
+
+  auto EntrySymbolIt = DIEOffsetToAccelEntryLabel.find(Entry.getDieOffset());
```

Comment:
With split dwarf we can have multiple DIEs with the same offset. Won't this potentially return the wrong label, or am I missing something?

Reply:
That is a good question. Maybe the keys here need to be a pair of UnitID + DIEOffset?

Reply:
That should work

Reply:
Actually never mind. This will work for split-dwarf. Since there should be only one CU per module.
But I think collision is still possible with CU/TUs since offsets are relative to CU/TU, or with lto. I think multiple CUs can be in one module with that. 
Yeah I think UnitID should be combined with DIEOffset.

BTW there is a code that suppose to unique things, but I don't think it's working right now:
https://github.com/llvm/llvm-project/pull/74391
Need to circle back to it.

Reply:
Though IIUC, in AsmPrinter you'd have at most one .o and one .dwo file, so you may be able to get away with having just one extra bit for the disambiguation.

Reply:
Will work on integrating UnitID next

Reply:
@ayermolo @adrian-prantl can we assume the UnitID of the parent is always the same UnitID of the children?

Reply:
I think that was a silly question, a parent is a parent in the DIE tree, so they are in the same Unit by definition.
Please let me know if you think this is wrong